### DISCUSSION TITLE
Mismatch of transformer and cable losses between PGM and Vision

### DISCUSSION THREAD

**Discussion by NielsDirks** [PowerGridModel/.github#13] [Q&A] [answered]
I imported a simple Vision file into PGM, which resulted in the input_data dictionary as shown below. As can be seen, the P and Q of the load are set to 0, so the transformer is not loaded. The no-load power loss 'p0' of the transformer is 28850 Watt. When I run the power flow calculation in Vision, the loss in the transformer is 32.1 kW, however, with PGM this is 23.88 kW (as shown below with 'output_data'). Does anyone have any idea where this difference may come from? Furthermore, how can the transformer losses (P_in - P_out) be lower than the no-load losses?

For this example, the delta between Vision and PGM seems to be a constant 8.2 kW not depending on the (active power) load. I have the same issue for a model with just one cable. There, the delta between PGM and Vision is about 0.5 kW, not depending on the load.

Thanks in advance!

```
input_data = {'node': array([(0,  20000.), (1, 150000.)], dtype={'names': ['id', 'u_rated'], 'formats': ['<i4', '<f8'], 'offsets': [0, 8], 'itemsize': 16, 'aligned': True}),
'transformer': array([(2, 1, 0, 1, 1, 158000., 23200., 85000000., 0.2775, 299000., 0.01470246, 28850., 1, 2, 5, 0, 17, 33, 1, 20, 2250., 0.2775, 0.2775, 299000., 299000., 0., 0., nan, nan)], dtype={'names': ['id', 'from_node', 'to_node', 'from_status', 'to_status', 'u1', 'u2', 'sn', 'uk', 'pk', 'i0', 'p0', 'winding_from', 'winding_to', 'clock', 'tap_side', 'tap_pos', 'tap_min', 'tap_max', 'tap_nom', 'tap_size', 'uk_min', 'uk_max', 'pk_min', 'pk_max', 'r_grounding_from', 'x_grounding_from', 'r_grounding_to', 'x_grounding_to'], 'formats': ['<i4', '<i4', '<i4', 'i1', 'i1', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', 'i1', 'i1', 'i1', 'i1', 'i1', 'i1', 'i1', 'i1', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8'], 'offsets': [0, 4, 8, 12, 13, 16, 24, 32, 40, 48, 56, 64, 72, 73, 74, 75, 76, 77, 78, 79, 80, 88, 96, 104, 112, 120, 128, 136, 144], 'itemsize': 152, 'aligned': True}),
'source': array([(3, 1, 1, 1., nan, 1.e+16, 0., 1.)], dtype={'names': ['id', 'node', 'status', 'u_ref', 'u_ref_angle', 'sk', 'rx_ratio', 'z01_ratio'], 'formats': ['<i4', '<i4', 'i1', '<f8', '<f8', '<f8', '<f8', '<f8'], 'offsets': [0, 4, 8, 16, 24, 32, 40, 48], 'itemsize': 56, 'aligned': True}),
'sym_load': array([(4, 0, 1, 0, 0., 0.)], dtype={'names': ['id', 'node', 'status', 'type', 'p_specified', 'q_specified'], 'formats': ['<i4', '<i4', 'i1', 'i1', '<f8', '<f8'], 'offsets': [0, 4, 8, 9, 16, 24], 'itemsize': 32, 'aligned': True})}

output_data = {'node': array([(0, 1, 1.05399568,  21079.91350826, -2.61801507e+00, 2.79492722e-08, -1.84544704e-08),
       (1, 1, 1.        , 149999.99998448, -2.38800960e-12, 2.38800960e+04,  1.03462448e+06)], dtype={'names': ['id', 'energized', 'u_pu', 'u', 'u_angle', 'p', 'q'], 'formats': ['<i4', 'i1', '<f8', '<f8', '<f8', '<f8', '<f8'], 'offsets': [0, 4, 8, 16, 24, 32, 40], 'itemsize': 48, 'aligned': True}),
'transformer': array([(2, 1, 0.01217529, 23880.09599732, 1034624.48076313, 3.98333208, 1034900.03149059, 2.79492722e-08, -1.84544704e-08, 9.17306727e-13, 3.34922273e-08)], dtype={'names': ['id', 'energized', 'loading', 'p_from', 'q_from', 'i_from', 's_from', 'p_to', 'q_to', 'i_to', 's_to'], 'formats': ['<i4', 'i1', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8', '<f8'], 'offsets': [0, 4, 8, 16, 24, 32, 40, 48, 56, 64, 72], 'itemsize': 80, 'aligned': True}),
'source': array([(3, 1, 23880.09599734, 1034623.49787923, 3.9833283, 1034899.04886839, 0.02307481)], dtype={'names': ['id', 'energized', 'p', 'q', 'i', 's', 'pf'], 'formats': ['<i4', 'i1', '<f8', '<f8', '<f8', '<f8', '<f8'], 'offsets': [0, 4, 8, 16, 24, 32, 40], 'itemsize': 48, 'aligned': True}),
'sym_load': array([(4, 1, 0., 0., 0., 0., 0.)], dtype={'names': ['id', 'energized', 'p', 'q', 'i', 's', 'pf'], 'formats': ['<i4', 'i1', '<f8', '<f8', '<f8', '<f8', '<f8'], 'offsets': [0, 4, 8, 16, 24, 32, 40], 'itemsize': 48, 'aligned': True})}
```

**Comment by TonyXiang8787** [comment]
Dear @NielsDirks, thanks for reaching us. Can you export the PGM input and output data using our export function as shown in the [documentation](https://power-grid-model.readthedocs.io/en/stable/examples/Serialization%20Example.html)?

Also, would it be possible to upload the `vnf` Vision file you used in the comparison?

**Reply by NielsDirks** [reply]
Dear @TonyXiang8787,

I used the 'json_serialize_to_file()' function, which resulted in the following files:
[input_data_from_vision.json](https://github.com/PowerGridModel/.github/files/13503100/input_data_from_vision.json)
[input_data_no_load.json](https://github.com/PowerGridModel/.github/files/13503102/input_data_no_load.json)
[output_data.json](https://github.com/PowerGridModel/.github/files/13503103/output_data.json)

The only change between input_data_no_load and input_data_from_vision is that p_specified of the load is set to 0 in the no_load file. This file is used to generate the output. Furthermore, below you can find the Vision file that was used. The vnf file is not supported for upload here, so I attached the exported excel file:
[TransformerOnly.xlsx](https://github.com/PowerGridModel/.github/files/13503130/TransformerOnly.xlsx)

If you need anything else, please let me know! Thanks

**Reply by mgovers** [reply]
@nitbharambe is investigating.

@nitbharambe can you either create a GitHub issue and/or post your updates here?

**Reply by nitbharambe** [reply]
@NielsDirks , The investigation is finished and we need some time to process the results and present it. Will let you know when done.

**Reply by nitbharambe** [reply, marked as answer]
The investigation is done. The results that you posted here from PGM, ie. output_data is theoretically correct. We verified the PGM results with hand calculations and with OpenDSS. Hence you can move ahead with your analysis.
